OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -283,11 +283,29 @@ bool CanMoveFreelyInto(Actor actor, CPos cell, SubCell subCell, BlockedByActor c
 			}
 
 			var otherActors = subCell == SubCell.FullCell ? world.ActorMap.GetActorsAt(cell) : world.ActorMap.GetActorsAt(cell, subCell);
-			foreach (var otherActor in otherActors)
-				if (IsBlockedBy(actor, otherActor, ignoreActor, ignoreSelf, cell, check, cellFlag))
-					return false;
 
-			return true;
+			if (ignoreSelf)
+			{
+				// Any actor blocking us will prevent our movement, *unless* we are one of those actors.
+				var isBlocked = false;
+				foreach (var otherActor in otherActors)
+				{
+					if (actor == otherActor)
+						return true;
+					isBlocked = isBlocked || IsBlockedBy(actor, otherActor, ignoreActor, cell, check, cellFlag);
```

Comment:
Curious. Why is it needed to check all actors in the cell?
If actor A does not block actor self in subcell X, do you then still need to check actor B? Wouldn't this suggest that two actors can cover one subcell? Is this possible?

Reply:
In the scenario we're interested in `subCell` is set to `SubCell.FullCell` - so it's the whole cell and no a specific subcell.

With that in mind, consider the example with 5 infantry all in one cell (but each in a different subcell). All of these will block us in a normal situation.

We need the loop to check against every actor though, because if e.g. the 5th actor is "us" (i.e. `actor == otherActor`) then we want to consider the cell *unblocked* because we're in it.

Hence why the loop has to check every actor. Even if we know we're blocked by one of the earlier actors, we can later override that decision and decide to be unblocked, if it turns out we're one of those actors.